library/src/main/java/com/ricohapi/mstorage/MediaStorage.java

```diff
@@ -215,11 +224,68 @@ public void meta(String mediaId, String fieldName, CompletionHandler<Map<String,
     }
 
     public void addMeta(String mediaId, Map<String, String> userMeta, CompletionHandler<Object> handler) {
-        // TODO
+
+        try {
+            if (accessToken == null) {
+                throw new RicohAPIException(0, "wrong usage: use the connect method to get an access token.");
+            }
+            RicohAPIRequest request;
+
+            for(String userMetaKey : userMeta.keySet()) {
+                String requestUserMetaKey = replaceUserMeta(userMetaKey);
+                String value = userMeta.get(userMetaKey);
+                if (requestUserMetaKey == null || isValidValue(value)) {
+                    throw new RicohAPIException(0, "invalid parameter: " + "{" + userMetaKey + "=" + value + "}");
+                }
+                request = new RicohAPIRequest(ENDPOINT + "/" + mediaId + USER_META_PATH + "/" + requestUserMetaKey);
+                Map<String, String> header = new HashMap<>();
+                header.put("Authorization", "Bearer " + accessToken);
+                header.put("Content-Type", "text/plain");
+
+                request.put(header, value);
+
+                if (request.isSucceeded()) {
+                    handler.onCompleted(new Object());
+                } else {
+                    throw new RicohAPIException(request.getResponseCode(), request.getErrorBody());
+                }
+            }
+        } catch (IOException e) {
+            handler.onThrowable(e);
+        } catch (RicohAPIException e) {
+            handler.onThrowable(e);
+        }
     }
 
     public void removeMeta(String mediaId, String key, CompletionHandler<Object> handler) {
         // TODO
     }
 
+    private String replaceUserMeta(String userMeta){
+        Pattern pattern = Pattern.compile(REPLACE_USER_META_REGEX);
+        Matcher match = pattern.matcher(userMeta);
+
+        String result;
+
+        if(match.matches()){
+            result = match.group(FIRST);
```

Comment:
I think this can be written `return match.group`.
It will be not to need to declare `result`.